contracts/pre-propose/cwd-pre-propose-approval-single/src/contract.rs

```diff
@@ -0,0 +1,370 @@
+    } else {
+        vec![]
+    };
+
+    let approval_id = advance_approval_id(deps.storage)?;
+
+    let propose_msg_internal = match msg {
+        ProposeMessage::Propose {
+            title,
+            description,
+            msgs,
+        } => ProposeMsg {
+            title,
+            description,
+            msgs,
+            proposer: Some(info.sender.into_string()),
+        },
+    };
+
+    // Prepare proposal submitted hooks msg to notify approver.  Make
+    // a proposal on the approver DAO to approve this pre-proposal
+
+    // TODO(zeke): is this actually what we should be feeding to the
+    // proposal submitted hooks? this locks them into being an
+    // approver.
+    let hooks_msgs =
+        pre_propose_base
+            .proposal_submitted_hooks
+            .prepare_hooks(deps.storage, |a| {
+                let execute_msg = WasmMsg::Execute {
+                    contract_addr: a.into_string(),
+                    msg: to_binary(&ExecuteBase::<ApproverProposeMessage, Empty>::Propose {
+                        msg: ApproverProposeMessage::Propose {
+                            title: propose_msg_internal.title.clone(),
+                            description: propose_msg_internal.description.clone(),
+                            approval_id,
+                        },
+                    })?,
+                    funds: vec![],
+                };
+                Ok(SubMsg::new(execute_msg))
+            })?;
+
+    // Save the proposal and its information as pending.
+    PENDING_PROPOSALS.save(
+        deps.storage,
+        approval_id,
+        &PendingProposal {
+            approval_id,
+            msg: propose_msg_internal,
+            deposit: config.deposit_info,
+        },
+    )?;
+
+    Ok(Response::default()
+        .add_messages(deposit_messages)
+        .add_submessages(hooks_msgs)
+        .add_attribute("method", "pre-propose")
+        .add_attribute("id", approval_id.to_string()))
+}
+
+pub fn execute_approve(
+    deps: DepsMut,
+    info: MessageInfo,
+    id: u64,
+) -> Result<Response, PreProposeError> {
+    // Check sender is the approver
+    let approver = APPROVER.load(deps.storage)?;
+    if approver != info.sender {
+        return Err(PreProposeError::Unauthorized {});
+    }
+
+    // Load proposal and send propose message to the proposal module
+    let proposal = PENDING_PROPOSALS.may_load(deps.storage, id)?;
+    match proposal {
+        Some(proposal) => {
+            let proposal_module = PrePropose::default().proposal_module.load(deps.storage)?;
+            let propose_messsage = WasmMsg::Execute {
+                contract_addr: proposal_module.into_string(),
+                msg: to_binary(&ProposeMessageInternal::Propose(proposal.msg))?,
+                funds: vec![],
+            };
+
+            PENDING_PROPOSALS.remove(deps.storage, id);
+            DEPOSIT_SNAPSHOT.save(deps.storage, &proposal.deposit)?;
```

Comment:
would be great to add comment about why this is needed

Reply:
yeah :) this was a failed design attempt. see: https://github.com/DA0-DA0/dao-contracts/issues/553

Reply:
the solution is to add a query to the proposal module interface (which was introduced in the macros PR) for `NextProposalId {}`. Having that, the trouble where the pre-propose module has to wait for the hook to know the ID will go away as it can query that and immediately create the proposal.